elfo-network/src/socket/raw/tcp.rs

```diff
@@ -1,12 +1,64 @@
-use std::net::SocketAddr;
+use std::{net::SocketAddr, sync::Arc};
 
 use derive_more::Display;
 use eyre::{Result, WrapErr};
 use futures::Stream;
 use tokio::net::{TcpListener, TcpStream};
 use tracing::warn;
 
-pub(super) use tokio::net::tcp::{OwnedReadHalf, OwnedWriteHalf};
+pub(super) use tokio::net::tcp::OwnedReadHalf;
+
+use std::{
+    io::{IoSlice, Result as IoResult},
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tokio::io::AsyncWrite;
+
+pub(in crate::socket) struct OwnedWriteHalf {
+    inner: tokio::net::tcp::OwnedWriteHalf,
+    tcp_socket_ref: Option<Arc<std::net::TcpStream>>,
```

Comment:
You can access fd via `inner`
* https://docs.rs/tokio/latest/tokio/net/tcp/struct.OwnedReadHalf.html#impl-AsRef%3CTcpStream%3E-for-OwnedReadHalf
* https://docs.rs/tokio/latest/tokio/net/struct.TcpStream.html#impl-AsRawFd-for-TcpStream